components/add-layers/add-layers.component.js

```diff
@@ -80,15 +80,23 @@ export const HsAddLayersComponent = {
         HsLayoutService.setMainPanel('datasource_selector');
         $scope.type = type.toUpperCase();
         $timeout(() => {
-          $rootScope.$broadcast(`ows.${type}_connecting`, url);
+          if (type == 'wms') {
+            HsEventBusService.wmsConnecting.next({uri: url});
+          } else {
+            $rootScope.$broadcast(`ows.${type}_connecting`, url);
+          }
         });
       }
     }
 
-    HsEventBusService.owsFilling.subscribe((type, url, layer) => {
+    HsEventBusService.owsFilling.subscribe(({type, uri, layer}) => {
       $scope.type = type.toLowerCase();
       $timeout(() => {
-        $rootScope.$broadcast(`ows.${type}_connecting`, url, layer);
+        if (type == 'wms') {
+          HsEventBusService.wmsConnecting.next({uri, layer});
```

Comment:
I see what you mean. That is a good point. What I dislike about the previous implementation (which used template literals), is how hard it was to trace back which methods fire which type of {type}_connecting event. It would be now a bit more of a code, but a quicker insight, I think.
Also, with manually defined wmsConnecting, wfsConnecting, etc. observables, the app developer can see with a single sight which types are possibly available. With {type}Connecting, this is somewhat unbound and you need to crawl through quite some code to get any idea which types can happen.
Yet the same can be probably achieved with a proper documentation, so.. I am not strictly against it, just need to be convinced about it :-) Opinions about this matter @DailisLangovskis or @raitisbe ?

Reply:
FTR, it seems I have had a different reasoning when upgrading the add-layers module 😄 I refactored it into `owsConnecting` observable there, with `type` as its property. See https://github.com/hslayers/hslayers-ng/compare/datasources-facelift...issue-761/add-layers-ng9?expand=1#diff-70da3f5786c6ddb22db0e7a2a2ecb1f807bbf6178fae1739d41fcef7183a9f40

Reply:
owsConnecting  seems like even better approach to me as we dont have to create multiple observables. Id go with that